src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
```

Comment:
There is no need of casting the array to 'T[]'. 
Let's just create a field 'elementData' and initialize it in a constructor and then use it as a storage for our objects. 

It should look like this:
```suggestion
    private Object[] elementData;
    
    public ArrayList() {
        this.elementData = new Object[DEFAULT_CAPACITY];
    }
```